src/main/java/com/gildedrose/infraestructure/repositories/ItemsRepository.java

```diff
@@ -0,0 +1,66 @@
+package com.gildedrose.infraestructure.repositories;
```

Comment:
El nombre de la clase tendría que ser otro. Un Repository se dedica exclusivamente a salvar y no a gestionar lógica.

Reply:
Entonces si solo cambiamos el nombre de la clase arreglaría el problema?

Reply:
Hay algo que se podría mejorar. En esta clase has englobado toda la lógica del programa(los updates del quality, SellIn etc )  . Ahora mismo no es un problema "**_muy_**" grande porque hay solo 3 tipos de items, pero que pasaría si en un futuro en vez de 3 tipos de items tenemos 10? o 30 o 1000?, y cada uno con sus reglas específicas?.... Acabaría siendo una clase con miles de líneas de código, dificultando **_muchísimo_** los posibles cambios que queramos hacer, y aumentando el riesgo de bugs. Esto rompe con el principio de open to extension.

Reply:
daqweqw

Reply:
En efecto el nombre de la clase ha de cambiarse.
Teniendo en cuenta que todos los items tienen los mismos atributos y que tan solo es poner un metodo para ciertos objetos, ya que otros funcionan con el metodo por defecto. Me parece mas logico crear para estas clases con variacion en sus updates, susus metodos particulares, juntos en una misma clase que englobe estas excepciones.
Ya que crear clases tipo item iguales en las que solo cambia el nombre, para poner su update correspondiente en caso de tenerlo me parece que estariamos repitiendo demasiado codigo inecesariamente ya que solo necesitamos un update.
Por otro lado por cada objeto nuevo se crea su excepcion si la hay y da igual cuantas haya en esta clase que engloba todas ya que la llamaremos siempre desde nuestra clase principal la cual llamara al metodo correspondiente para cada clase item. La modificacion de ser necesaria es sencialla porque para cada excepcion tenemos su nombre espedifico y unico.